### PR TITLE
Sandeepyadav1478/2025 06 19 deberta v2 model card update

### PR DESCRIPTION
**Pull request by sandeepyadav1478**
# What does this PR do?

Updates the DeBERTa-v2 card according to the description in [#36979](https://github.com/huggingface/transformers/issues/36979) to standardize all model cards' look.
The pr includes:
-  Model description
-  Examples for Pipeline, AutoModel, and command line usage

Note: AttentionMaskVisualizer is not applicable either, as DeBERTa-v2is currently not supported

- [x] This PR improves the docs

Who can review?

@stevhliu Please have a look and let me know if any further work is needed. thank you

### CODE REVIEW

File: docs/source/en/model_doc/deberta-v2.md [hunk 1]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+</div>
+
+
 # DeBERTa-v2
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
-</div>
+[DeBERTa-v2](https://huggingface.co/papers/2006.03654) is an advanced language model that builds on the original DeBERTa architecture, which itself is an evolution of BERT and RoBERTa. What makes DeBERTa-v2 special is its unique way of handling the meaning and position of words separately, using a disentangled attention mechanism—this helps it better understand context and relationships in language. The v2 version introduces a new, larger vocabulary, a sentencepiece-based tokenizer, nGram Induced Input Encoding (nGiE) for better local dependency learning, and more efficient parameter sharing in the attention layers. It also comes in larger sizes (900M and 1.5B parameters), which means it can tackle even more complex language tasks with higher accuracy.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]: Only need to focus on v2 :)

```suggestion
[DeBERTa-v2](https://huggingface.co/papers/2006.03654) improves on the original [DeBERTa](./deberta) architecture by using a SentencePiece-based tokenizer and a new vocabulary size of 128K. It also adds an additional convolutional layer within the first transformer layer to better learn local dependencies of input tokens. Finally, the position projection and content projection matrices are shared in the attention layer to reduce the number of parameters.
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 2]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+
+You can find all the original [DeBERTa-v2] checkpoints under the [DeBERTa-v2](https://huggingface.co/microsoft?search_models=deberta-v2) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
You can find all the original [DeBERTa-v2] checkpoints under the [Microsoft](https://huggingface.co/microsoft?search_models=deberta-v2) organization.
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 3]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+
+
+> [!TIP]
+> Click on the [DeBERTa-v2] models in the right sidebar for more examples of how to apply [DeBERTa-v2] to different text classification, token classification, question answering, and multiple choice tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
> [!TIP]
> This model was contributed by [Pengcheng He](https://huggingface.co/DeBERTa).
>
> Click on the [DeBERTa-v2] models in the right sidebar for more examples of how to apply [DeBERTa-v2] to different language tasks.
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 4]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md:43]: Use mnli finetuned checkpoint

```py
import torch
from transformers import pipeline
pipeline = pipeline(
    task="text-classification",
    model="microsoft/deberta-v2-xlarge-mnli",
    device=0,
    torch_dtype=torch.float16
)
pipeline("DeBERTa-v2 is great at understanding context!")
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 5]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+)
+result = classifier("DeBERTa-v2 is great at understanding context!")
+print(result)
+```
+
+</hfoption>
+<hfoption id="AutoModel">
+
+```py
+from transformers import AutoTokenizer, AutoModel
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```py
import torch
from transformers import AutoTokenizer, AutoModelForSequenceClassification

tokenizer = AutoTokenizer.from_pretrained(
    "microsoft/deberta-v2-xlarge-mnli",
)
model = AutoModelForSequenceClassification.from_pretrained(
    "microsoft/deberta-v2-xlarge-mnli",
    torch_dtype=torch.float16,
    device_map="auto",
)
inputs = tokenizer("DeBERTa-v2 is great at understanding context!", return_tensors="pt").to("cuda")
outputs = model(**inputs)

logits = outputs.logits
predicted_class_id = logits.argmax().item()
predicted_label = model.config.id2label[predicted_class_id]
print(f"Predicted label: {predicted_label}")
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 6]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+
+inputs = tokenizer("DeBERTa-v2 is great at understanding context!", return_tensors="pt")
+outputs = model(**inputs)
+print(outputs.last_hidden_state)
+
+```
+
+</hfoption>
+
+<hfoption id="transformers-cli">
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
<hfoption id="transformers CLI">
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 7]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+
+```bash
+echo -e "Plants create [MASK] through a process known as photosynthesis." | transformers-cli run --task fill-mask --model microsoft/deberta-v2-xlarge --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
echo -e "DeBERTa-v2 is great at understanding context!" | transformers-cli run --task text-classification --model microsoft/deberta-v2-xlarge-mnli --device 0
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 8]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+```
+</hfoption>
+</hfoptions>
+
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
+
+The example below uses [bitsandbytes quantization](https://huggingface.co/docs/transformers/main/en/main_classes/quantization) to only quantize the weights to 8-bit.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]: Remember to add code example under here!

```suggestion
The example below uses [bitsandbytes](../quantization/bitsandbytes) to only quantize the weights to 4-bits.
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 9]
```diff
@@ -14,58 +14,83 @@ rendered properly in your Markdown viewer.
...
+
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]: Can remove these notes since they're addressed in the beginning

**Review by stevhliu — COMMENTED** [review]
Thanks for your contribution!

**Comment by sandeepyadav1478** [comment]
Done with changes you pointed out @stevhliu, Thanks for letting me work on it.
Lmk, if I can help with any other issue. 
As, I'm new to this repo., I might take little extra time but really curious to work.

File: docs/source/en/model_doc/deberta-v2.md (hunk 8 above)
Comment (stevhliu) [reply docs/source/en/model_doc/deberta-v2.md]: Hey, this is unresolved. It would be super nice to have a code example of quantization here

File: docs/source/en/model_doc/deberta-v2.md (hunk 6 above)
Comment (stevhliu) [reply docs/source/en/model_doc/deberta-v2.md]: Should just be `transformers CLI` instead of `transformers-CLI`

File: docs/source/en/model_doc/deberta-v2.md [hunk 10]
```diff
@@ -14,58 +14,85 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
+
+<hfoptions id="usage">
+<hfoption id="Pipeline">
+
+```py
+import torch
+from transformers import pipeline
+
+# Example: Text classification with DeBERTa-v2
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 11]
```diff
@@ -14,58 +14,85 @@ rendered properly in your Markdown viewer.
...
+classifier = pipeline(
+    task="text-classification",
+    model="microsoft/deberta-v2-xlarge-mnli",
+    device=0,
+    torch_dtype=torch.float16
+)
+result = classifier("DeBERTa-v2 is great at understanding context!")
+print(result)
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
pipeline = pipeline(
    task="text-classification",
    model="microsoft/deberta-v2-xlarge-mnli",
    device=0,
    torch_dtype=torch.float16
)
pipeline("DeBERTa-v2 is great at understanding context!")
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 12]
```diff
@@ -14,58 +14,85 @@ rendered properly in your Markdown viewer.
...
+
+You can find all the original [DeBERTa-v2] checkpoints under the [Microsoft](https://huggingface.co/microsoft?search_models=deberta-v2) organization.
+
+
+> [!TIP]
+> This model was contributed by [Pengcheng He](https://huggingface.co/DeBERTa).
+>
+> Click on the [DeBERTa-v2] models in the right sidebar for more examples of how to apply [DeBERTa-v2] to different language tasks.
+
+The example below demonstrates how to generate text based on an image with [`Pipeline`] or the [`AutoModel`] class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
The example below demonstrates how to classify text with [`Pipeline`] or the [`AutoModel`] class.
```

**Review by stevhliu — COMMENTED** [review]
Just a few more comments and then we should be ready to merge!

File: docs/source/en/model_doc/deberta-v2.md (hunk 12 above)
Comment (sandeepyadav1478) [reply docs/source/en/model_doc/deberta-v2.md]: Nice catch

File: docs/source/en/model_doc/deberta-v2.md (hunk 8 above)
Comment (sandeepyadav1478) [reply docs/source/en/model_doc/deberta-v2.md]: Done

**Comment by sandeepyadav1478** [comment]
Hey @stevhliu,

Done with changes, lmk for else.
Thanks

File: docs/source/en/model_doc/deberta-v2.md [hunk 13]
```diff
@@ -14,58 +14,113 @@ rendered properly in your Markdown viewer.
...
-</div>
+[DeBERTa-v2](https://huggingface.co/papers/2006.03654) improves on the original [DeBERTa](./deberta) architecture by using a SentencePiece-based tokenizer and a new vocabulary size of 128K. It also adds an additional convolutional layer within the first transformer layer to better learn local dependencies of input tokens. Finally, the position projection and content projection matrices are shared in the attention layer to reduce the number of parameters.
+
+You can find all the original [DeBERTa-v2] checkpoints under the [Microsoft](https://huggingface.co/microsoft?search_models=deberta-v2) organization.
+
+
+> [!TIP]
+> This model was contributed by [Pengcheng He](https://huggingface.co/DeBERTa).
+>
+> Click on the [DeBERTa-v2] models in the right sidebar for more examples of how to apply [DeBERTa-v2] to different language tasks.
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]:
```suggestion
> Click on the DeBERTa-v2 models in the right sidebar for more examples of how to apply DeBERTa-v2 to different language tasks.
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 14]
```diff
@@ -14,58 +14,113 @@ rendered properly in your Markdown viewer.
...
+)
+
+text = "Hugging Face Transformers make NLP easy!"
+inputs = tokenizer(text, return_tensors="pt")
+inputs = {k: v.to(model.device) for k, v in inputs.items()}
+
+outputs = model(**inputs)
+logits = outputs.logits
+
+print("Logits:", logits)
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]: For consistency, best to maintain the same example across the board

```suggestion
inputs = tokenizer("DeBERTa-v2 is great at understanding context!", return_tensors="pt").to("cuda")
outputs = model(**inputs)

logits = outputs.logits
predicted_class_id = logits.argmax().item()
predicted_label = model.config.id2label[predicted_class_id]
print(f"Predicted label: {predicted_label}")
```

File: docs/source/en/model_doc/deberta-v2.md [hunk 15]
```diff
@@ -14,58 +14,113 @@ rendered properly in your Markdown viewer.
...
-  experiments, this can save parameters without affecting the performance.
-- **Apply bucket to encode relative positions** The DeBERTa-v2 model uses log bucket to encode relative positions
-  similar to T5.
-- **900M model & 1.5B model** Two additional model sizes are available: 900M and 1.5B, which significantly improves the
-  performance of downstream tasks.
-
-This model was contributed by [DeBERTa](https://huggingface.co/DeBERTa). This model TF 2.0 implementation was
-contributed by [kamalkraj](https://huggingface.co/kamalkraj). The original code can be found [here](https://github.com/microsoft/DeBERTa).
 
 ## Resources
```
Comment (stevhliu) [inline docs/source/en/model_doc/deberta-v2.md]: You can remove the `## Resources` section as they are only internal links

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more changes!

**Comment by sandeepyadav1478** [comment]
@stevhliu Done with :arrow_up: changes.
Can you assign me new task please?

Wanna join any technical task :relaxed:
